src/shogun/features/DotFeatures.cpp

```diff
@@ -81,7 +78,7 @@ void CDotFeatures::dense_dot_range(float64_t* output, int32_t start, int32_t sto
 #endif
 
 		int32_t t_start=thread_num*step;
-		int32_t t_stop=(thread_num==num_threads) ? stop : (thread_num+1)*step;
+		int32_t t_stop=(thread_num==num_threads) ? num_vectors : (thread_num+1)*step;
 
 #ifdef WIN32
 		for (int32_t i=t_start; i<t_stop; i++)
```

Comment:
Can the TODO below be removed?

Reply:
Should we also remove the progress bar? I don't think such an operation needs it.

Reply:
Why not? It could be useful if many vectors dotted.

Reply:
But does the user need to see them? I think the progress bar would make sense if it indicated the state of the algorithm, instead of an operation.

Reply:
About the TODO. Do you mean remove it with the signal stuff, or implement it? If so, I would need to first check out what are these signals about :"D